app/controllers/spree/api/v2/storefront/product_questions_controller.rb

```diff
@@ -17,7 +17,7 @@ def show
           def create
             @product_question = Spree::ProductQuestion.new(product_question_params)
             @product_question.product = @product
-            @product_question.user = spree_current_user if spree_user_signed_in?
+            @product_question.user = spree_current_user
```

Comment:
spree_current_user
have you ensured we don't get some 401 error or redirect in case user is not logged in?

having rspect test for this controller would make us more confident about this :)